### ISSUE TITLE
[3.0] FormModelDescriber doesn't go down recursive in forms. 

### ISSUE THREAD

**Issue by kopaygorodsky** [nelmio/NelmioApiDocBundle#1079]
Only main form is handled. Is it expected behavior? 
If yes, then I have a question how to describe the full form with all child forms.

**Comment by kopaygorodsky** [comment]
As I understood I can pass FormType to `@Model ` annotation and it will be parsed&defined as Model. 
I played a bit with bundle and force it to recognize embedded forms. 
In result I have an array of Definitions where are Entities and Forms together(like User and UserType).  Is it right way of implementing this? 


I added model registry trait to FormModelDescriber and at the end of second cycle 
```
$model = new Model(new Type(Type::BUILTIN_TYPE_OBJECT, false, FORM_TYPE_CLASS));
                    $property->setRef(
                        $this->modelRegistry->register($model)
                    );
```
If it's ok, I can work in this way deeper and contribute my solution.
